src/components/WithScroll/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from "react";
+
+const WithScroll = ComposedComponent =>
+  class ScrollDecorator extends Component {
+    constructor() {
+      super();
+
+      this.state = {
+        scrollPosition: this.getWindowScrollTop()
+      };
+
+      this.handleInterval = this.handleInterval.bind(this);
```

Comment:
you don't need bind if defined handleInterval as arrow function

```
  handleInterval=()=> {
      cancelAnimationFrame(this.requestID);
      this.requestID = requestAnimationFrame(this.handleRequestAnimationFrame);
    }
```
so this line you don't need
```
      this.handleInterval = this.handleInterval.bind(this);
```